### PR TITLE
Featurebranch dkw1263 config from db

### PR DESCRIPTION
**Pull request by JanPeterDatakind**
Configuration for projects is now pulled from the DB instead of dot_config.yml file. If the project has not been defined, DOT stops and informs the user immediately instead of after the run. If the project is defined in the dot_config.yml file, the information is NOT taken from the database but from the dot_config.yml file.

### CODE REVIEW

File: docker/dot/dot_config.yml [hunk 1]
```diff
@@ -9,12 +9,3 @@ dot_db:
   dbname: dot_db
   schema: dot
   threads: 4
-ScanProject1_db:
```
Comment (lrnzcig) [inline docker/dot/dot_config.yml:12]: need to double check, but I this is going to break self tests if you try to run it from docker
personally I don't use the feature, I always run self tests from local and from the execution of PRs in github
so I would agree to remove it, if we know what we are doing, and I would say let's remove it

File: dot/config/example/self_tests/dot_config_docker.yml [hunk 2]
```diff
@@ -10,12 +10,3 @@ dot_db:
   dbname: dot_db
   schema: self_tests_dot
   threads: 4
-ScanProject1_db:
```
Comment (lrnzcig) [inline dot/config/example/self_tests/dot_config_docker.yml:13]: same as above, I would say let's remove

File: dot/config/example/self_tests/dot_config_local.yml [hunk 3]
```diff
@@ -10,12 +10,3 @@ dot_db:
   dbname: dot_db
   schema: self_tests_dot
   threads: 4
-ScanProject1_db:
```
Comment (lrnzcig) [inline dot/config/example/self_tests/dot_config_local.yml:13]: this one I would not remove, used as a reference if you want to setup self tests locally

**Review by lrnzcig — COMMENTED** [review]
from the 3 comments that I left, I would only request changes for the 3rd, i.e. `dot/config/example/self_tests/dot_config_local.yml` should have the config for self_tests, since if you use that file it's because you want to set self tests locally... let me know if it does not make sense for you

for the rest this is a great PR, and I see you leave untouched the yml [in the workflow for the github action](https://github.com/datakind/Data-Observation-Toolkit/blob/main/.github/workflows/ci.yml#L35) to trigger self tests for github

thanks!

File: dot/config/example/self_tests/dot_config_docker.yml (hunk 2 above)
Comment (JanPeterDatakind) [reply dot/config/example/self_tests/dot_config_docker.yml:13]: Removed as suggested.

**Comment by JanPeterDatakind** [comment]
@lrnzcig thank you for the review! As you suggested, I reverted the change in 'dot/config/example/self_tests/dot_config_local.yml' and left 'dot_config_github.yml' and 'dot_config_docker.ym'l without 'ScanProject1'.

**Review by lrnzcig — APPROVED** [review]
thanks!

**Pull request merged**